src/Microsoft.DotNet.Interactive.ExtensionLab/NteractDataExplorerExtensions.cs

```diff
@@ -0,0 +1,91 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Text;
+using Microsoft.AspNetCore.Html;
+using Microsoft.DotNet.Interactive.Formatting;
+using Microsoft.DotNet.Interactive.Http;
+
+namespace Microsoft.DotNet.Interactive.ExtensionLab
+{
+    public static class NteractDataExplorerExtensions
+    {
+        public static TabularDataResourceExplorerView<NteractDataExplorer> ToView(this NteractDataExplorer explorer)
+        {
+            return new(explorer);
+        }
+
+        public static T UseNteractDataExplorer<T>(this T kernel, string uri = null, string context = null, string cacheBuster = null) where T : Kernel
+        {
+            RegisterFormatters(string.IsNullOrWhiteSpace(uri) ? null : new Uri(uri), context, cacheBuster);
+            return kernel;
+        }
+
+        private static void RegisterFormatters(Uri uri, string context, string cacheBuster)
+        {
+            Formatter.Register<TabularDataResourceExplorerView<NteractDataExplorer>>((explorerWidget, writer) =>
+            {
+                var html = explorerWidget.TabularDataResourceExplorer.RenderDataExplorer(uri,context,cacheBuster);
+                writer.Write(html);
+            }, HtmlFormatter.MimeType);
+        }
+
+        private static HtmlString RenderDataExplorer(this NteractDataExplorer explorer, Uri uri, string context, string cacheBuster)
+        {
+            var divId = explorer.Id;
+            var data = explorer.TabularDataResource.ToJson();
+            var code = new StringBuilder();
+            code.AppendLine("<div style=\"background-color:white;\">");
+            code.AppendLine($"<div id=\"{divId}\" style=\"height: 100ch ;margin: 2px;\">");
+            code.AppendLine("</div>");
+            code.AppendLine(@"<script type=""text/javascript"">");
+            GenerateCode(data, code, divId, "https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.6/require.min.js", uri, context, cacheBuster);
+            code.AppendLine(" </script>");
+            code.AppendLine("</div>");
+            return new HtmlString(code.ToString());
+        }
+
+        private static void GenerateCode(TabularDataResourceJsonString data, StringBuilder code, string divId, string requireUri, Uri uri, string context, string cacheBuster)
```

Comment:
`Generate*` implies that these methods perform some side effect. The intention of the code can be made clearer by refactoring them to extension methods on the `StringBuilder`, e.g. `AppendRequireLoader`, etc.